hot-desk/src/main/java/com/exadel/sandbox/booking/service/BookingService.java

```diff
@@ -27,4 +27,67 @@
     public BookingService(ModelMapper mapper, BookingRepository repository) {
         super(mapper, repository);
     }
+    public final static int MAX_MONTH = 3;
+
+
+    @Override
+    public ResponseEntity<BookingResponseDto> create(BookingCreateDto bookingCreateDto) {
+        Booking booking = mapper.map(bookingCreateDto, Booking.class);
+
+        checkNewBooking(booking);
+
+        BookingResponseDto bookingResponseDto = mapper.map(repository.save(booking), BookingResponseDto.class);
+
+        return new ResponseEntity<>(bookingResponseDto, HttpStatus.CREATED);
+    }
+
+    //    Throws exception if any of the objects already booked
+    private void checkNewBooking(Booking booking) {
+        List<LocalDate> dates = booking.getDates().stream().map(BookingDates::getDate).collect(Collectors.toList());
+
+        checkTheDates(dates);
+
+        Integer integer = repository.checkIfSeatIsFree(booking.getSeat().getId(), dates);
+        if (integer == 0)
+            throwException("Seat " + booking.getSeat().getId());
+
+        if (repository.checkIfEmployeeNotBooked(booking.getEmployee().getId(), dates) == 0)
+            throwException("Employee " + booking.getEmployee().getId());
+
+        if (booking.getParkingSpot() != null)
+            if (repository.checkIfParkingSpotIsFree(booking.getParkingSpot().getId(), dates) == 0)
+                throwException("Parking Spot " + booking.getParkingSpot().getId());
+    }
+
+    //    Checks if Date is in valid range
+    private void checkTheDates(List<LocalDate> dates) {
+        LocalDate today = LocalDate.now();
+
+        LocalDate dateLimit = today.plusMonths(MAX_MONTH);
+
+        dates.forEach(date -> {
+            if (!(date.isAfter(today) && date.isBefore(dateLimit)) && !(date.equals(today) || date.equals(dateLimit)))
+                throw new DateOutOfBoundException("Date " + date + " is not in the range ");
+        });
+    }
+
+    //    Throws exception;
+    private void throwException(String object) {
+        throw new DoubleBookingInADayException(object + " already booked on these days");
+    }
+
+    public ResponseEntity<List<BookingResponseDto>> getCurrentBookings() {
+        LocalDate today = LocalDate.now();
+        LocalDate limitDates = today.plusMonths(MAX_MONTH);
+
+        List<Booking> bookingsInTimePeriod = repository.findBookingsInTimePeriod(today, limitDates);
+
+        return ResponseEntity.ok(mapper.map(bookingsInTimePeriod, new TypeToken<List<BookingResponseDto>>(){}.getType()));
```

Comment:
Why u need new TypeToken<List<BookingResponseDto>>(){}.getType()? 
Just ResponseEntity.ok(list) do not work?

Reply:
The list is the List of Booking Not bookingResponseDto, I'm first mapping it

Reply:
Why u do not map like this?:
bookingsInTimePeriod.strema().map(mapper::toVO)...

new TypeToken<List> - looks a little bit strange

Reply:
I did it for simplicity looks like it's the opposite of my thoughts